zebra/zebra_rib.c

```diff
@@ -1247,8 +1247,17 @@ static void rib_process(struct route_node *rn)
 			SET_FLAG(new_selected->flags, ZEBRA_FLAG_SELECTED);
 
 		if (old_selected) {
-			if (!new_selected)
-				redistribute_delete(p, src_p, old_selected);
+			/*
```

Comment:
Why wouldn't a point fix of `if (!new_selected || CHECK_FLAG(old_selected->status, ROUTE_ENTRY_REMOVED)) redistribute_delete(p, src_p, old_selected);` be sufficient here?

Reply:
that was my original "quick" fix - but the drawback there has to do with the replace semantic of zebra's redistribution. if a zapi client is registered for the new type, the type of 'new_selected', then it would receive a delete (for 'old_selected') followed eventually by an add for 'new'. that would be a change of behavior, and potentially generate more protocol updates. that seemed unnecessary, so I tried to have a version of the api that deals with this exact case. in this version, if a client is not going to see 'new_selected', it will get a delete operation. all other clients see the same behavior from zebra that they see now - one update, where 'new' implicitly replaces 'old'.